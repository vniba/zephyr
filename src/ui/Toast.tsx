import toast, { ToastBar, Toaster } from 'react-hot-toast';
import { IoCloseCircleSharp } from 'react-icons/io5';
import styled from 'styled-components';

const CloseButton = styled.button`
  background-color: transparent;
  border: none;
  color: var(--color-grey-900);
  font-size: 2rem;
  cursor: pointer;
  &:hover,
  &:active,
  &:focus {
    outline: none;
    color: var(--color-grey-700);
  }
`;
function Toast() {
  const dismissToast = (toastID: string) => {
    toast.dismiss(toastID);
  };
  return (
    <Toaster
      position='top-center'
      gutter={13}
      containerStyle={{ margin: '8px' }}
      toastOptions={{
        success: {
          duration: 3000,
        },
        error: {
          duration: 5000,
        },
        style: {
          fontSize: '1.5rem',
          maxWidth: '550px',
          padding: '1.5rem 2rem',
          backgroundColor: 'var(--color-grey-0)',
          color: 'var(--color-grey-700)',
        },
      }}>
      {t => (
        <ToastBar toast={t}>
          {({ icon, message }) => (
            <>
              {icon}
              {message}
              {t.type !== 'loading' && (
                <CloseButton onClick={() => dismissToast(t.id)}>
                  <IoCloseCircleSharp />
                </CloseButton>
              )}
            </>
          )}
        </ToastBar>
      )}
    </Toaster>
  );
}

export default Toast;
