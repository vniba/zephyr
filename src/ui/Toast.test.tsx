import { render } from '@testing-library/react';
import Toast from './Toast';

describe('Toast component', function () {
  it('should match default snapshot', function () {
    const { container } = render(<Toast />);
    expect(container).toMatchSnapshot();
  });
});
