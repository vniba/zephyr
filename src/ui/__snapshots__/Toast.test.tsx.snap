// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toast component > should match default snapshot 1`] = `
<div>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none; margin: 8px;"
  />
</div>
`;
